.docker/Dockerfile-CPU

```diff
@@ -4,18 +4,75 @@
 # flashlight       master     (git, CPU backend)
 # ==================================================================
 
-FROM flml/flashlight:cpu-base-consolidation-latest
-
-ENV MKLROOT="/opt/intel/mkl"
+FROM flml/flashlight:cpu-base-consolidation-latest as build
 
 # ==================================================================
 # flashlight with CPU backend
 # ------------------------------------------------------------------
 # Setup and build flashlight
-RUN mkdir /root/flashlight
+RUN mkdir /tmp/flashlight
+
+COPY . /tmp/flashlight
+
+RUN mkdir -p /tmp/flashlight/build && \
+    cd /tmp/flashlight/build && \ 
+    cmake -DCMAKE_BUILD_TYPE="Release" -DCMAKE_INSTALL_PREFIX="/opt/flashlight" -DFL_BACKEND="CPU" -DFL_LIBRARIES_USE_CUDA="OFF" -DFL_BUILD_TESTS="ON" -DDNNL_DIR="/opt/onednn/lib/cmake/dnnl" -DGloo_INCLUDE_DIR="/opt/gloo/include" -DGloo_NATIVE_LIBRARY="/opt/gloo/lib/libgloo.a" .. && \
+    make -j$(nproc) && \
+    make install -j$(nproc)
+
+
 
-COPY . /root/flashlight
+#############################################################################
+#                            SECOND STAGE                                   #
+#############################################################################
 
-RUN cd /root/flashlight && mkdir -p build && \
-    cd build && cmake .. -DFL_BACKEND=CPU -DFL_LIBRARIES_USE_CUDA=OFF && \
-    make -j$(nproc) && make install
+FROM ubuntu:18.04
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+# install dependencies
+RUN apt-get update && apt-get install -y --no-install-recommends \
+        g++ \
+        # for cmake
+        zlib1g-dev libcurl4-openssl-dev \
+        # for MKL
+        apt-transport-https \
+        gpg-agent gnupg2 \
+        # for arrayfire CPU backend
+        # OpenBLAS
+        libopenblas-dev libfftw3-dev liblapacke-dev \
+        # ATLAS
+        libatlas3-base libatlas-base-dev libfftw3-dev liblapacke-dev \
+        # ssh for OpenMPI
+        openssh-server openssh-client \
+        # OpenMPI
+        libopenmpi-dev libomp-dev openmpi-bin \
+        # libsndfile
+        libsndfile1-dev \
+        # for libsndfile for ubuntu 18.04
+        libopus-dev \
```

Comment:
are we sure `libopus` is needed? I think the symbols `libsndfile1-dev` expects are already in `libogg` which is a transitive libsndfile dependency

Reply:
Let me recheck this, but when I tried on ubuntu 18 libopus was not in the list of dependencies and apt-packages which are installed with libsndfile1.

Reply:
```
root@176d6b2a8ddb:/# apt-get install libsndfile1-dev
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following additional packages will be installed:
  libdpkg-perl libfile-fcntllock-perl libflac-dev libflac8 libgdbm-compat4 libgdbm6 libglib2.0-0 libglib2.0-data liblocale-gettext-perl libogg-dev libogg0 libperl5.30 libsndfile1 libvorbis-dev
  libvorbis0a libvorbisenc2 libvorbisfile3 netbase perl perl-modules-5.30 pkg-config shared-mime-info xdg-user-dirs xz-utils
```